transport/httptransport/http_transport.go

```diff
@@ -0,0 +1,241 @@
+package httptransport
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sync"
+
+	"github.com/filecoin-project/boost/transport/types/transferstatus"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/libp2p/go-eventbus"
+	"github.com/libp2p/go-libp2p-core/event"
+
+	"github.com/filecoin-project/boost/transport/httptransport/pb"
+	"github.com/filecoin-project/boost/transport/types"
+	"github.com/gogo/protobuf/proto"
+
+	"github.com/filecoin-project/boost/transport"
+)
+
+var log = logging.Logger("http-transport")
+
+const (
+	readBufferSize = 4096
+)
+
+var _ transport.Transport = (*httpTransport)(nil)
+
+type httpTransport struct {
+	// TODO Construct and inject an http client here ?
+}
+
+func (h *httpTransport) Execute(ctx context.Context, transportInfo []byte, dealInfo *types.TransportDealInfo) (th transport.Handler, isComplete bool, err error) {
```

Comment:
what lets us signal which transport to use? this is assuming that transportInfo already knows it's using HTTP transport, so is there signalling a layer up of which transport to use?